core/src/test/java/kafka/admin/ConfigCommandUnitTest.java

```diff
@@ -0,0 +1,421 @@
+            shortFlag, "1",
+            "--alter",
+            "--add-config", "a;c=b"));
+
+        assertThrows(IllegalArgumentException.class,
+            () -> ConfigCommand.parseConfigsToBeAdded(inValidCreateOpts));
+
+        ConfigCommand.ConfigCommandOptions inValidCreateOpts2 = new ConfigCommand.ConfigCommandOptions(toArray(connectOpts1, connectOpts2,
+            shortFlag, "1",
+            "--alter",
+            "--add-config", "a,=b"));
+
+        assertThrows(IllegalArgumentException.class,
+            () -> ConfigCommand.parseConfigsToBeAdded(inValidCreateOpts2));
+    }
+
+    @Test
+    public void shouldFailIfAddAndAddFile() {
+        // Should not parse correctly
+        ConfigCommand.ConfigCommandOptions createOpts = new ConfigCommand.ConfigCommandOptions(toArray("--bootstrap-server", "localhost:9092",
+            "--entity-name", "1",
+            "--entity-type", "brokers",
+            "--alter",
+            "--add-config", "a=b,c=d",
+            "--add-config-file", "/tmp/new.properties"
+        ));
+        assertThrows(IllegalArgumentException.class, createOpts::checkArgs);
+    }
+
+    @Test
+    public void testParseConfigsToBeAddedForAddConfigFile() throws IOException {
+        String fileContents =
+            "a=b\n" +
+            "c = d\n" +
+            "json = {\"key\": \"val\"}\n" +
+            "nested = [[1, 2], [3, 4]]";
+
+        File file = TestUtils.tempFile(fileContents);
+
+        List<String> addConfigFileArgs = Arrays.asList("--add-config-file", file.getPath());
+
+        ConfigCommand.ConfigCommandOptions createOpts = new ConfigCommand.ConfigCommandOptions(toArray(Arrays.asList("--bootstrap-server", "localhost:9092",
+                "--entity-name", "1",
+                "--entity-type", "brokers",
+                "--alter"),
+            addConfigFileArgs));
+        createOpts.checkArgs();
+
+        Properties addedProps = ConfigCommand.parseConfigsToBeAdded(createOpts);
+        assertEquals(4, addedProps.size());
+        assertEquals("b", addedProps.getProperty("a"));
+        assertEquals("d", addedProps.getProperty("c"));
+        assertEquals("{\"key\": \"val\"}", addedProps.getProperty("json"));
+        assertEquals("[[1, 2], [3, 4]]", addedProps.getProperty("nested"));
+    }
+
+    @SuppressWarnings("deprecation") // Added for Scala 2.12 compatibility for usages of JavaConverters
+    public void testExpectedEntityTypeNames(List<String> expectedTypes, List<String> expectedNames, List<String> connectOpts, String...args) {
+        ConfigCommand.ConfigCommandOptions createOpts = new ConfigCommand.ConfigCommandOptions(toArray(Arrays.asList(connectOpts.get(0), connectOpts.get(1), "--describe"), Arrays.asList(args)));
+        createOpts.checkArgs();
+        assertEquals(createOpts.entityTypes().toSeq(), ConfigCommandIntegrationTest.seq(expectedTypes));
```

Comment:
Could we replace those scala collection by java collection in this PR? As we are in JDK8+, most lambda functions should work in Java. In order to reduce the size of PR, we rewrite only production code related to this PR. WDYT?

Reply:
Are you suggesting make `ConfigCommandOptions#entityTypes` return `java.util.List` instead of scala List?
I doubt it will make PR more clean, because `toSeq` and `seq` call are pretyy compact and self explaining.

We have full java version of Command waiting for a review after tests, also.
So, do we really need some intermediate steps?

Reply:
> We have full java version of Command waiting for a review after tests, also.
So, do we really need some intermediate steps?

I try to avoid large changes when we do "full java version of Command". Changing all collections from Scala to Java could result in another large "rewrite" for those tests. If those tests keep being "rewrite", what the point of separating those "smaller" patches?

I'm ok to keep current version if you feel above concern is false :)

Reply:
> Changing all collections from Scala to Java could result in another large "rewrite" for those tests.

First of all - I understand your point and agree with it.
I'm absolutely +1 for making patches as small and independent as possible to avoid any possible bugs during scala -> java migration.
Let me try to change collections from scala to java as another change before review of the command itself.
After we merge all tests in java.

Will it work for you?

> If those tests keep being "rewrite", what the point of separating those "smaller" patches?

To separate two reviews and eye checks - one for test itself - to make sure all checks are same in java and scala and I not removed some cases accidently.

Change of collection types must be similar to each other, so it will be easy to review separately.    

> I'm ok to keep current version if you feel above concern is false :)

Let's keep current version, please.